### PR TITLE
macOS CI

### LINKED ISSUE #65: [dev] macOS CI
I'll add this to the travis test matrix. We have quite a few macOS users.

### PR DESCRIPTION
**Pull request by hdhoang**
I'm not sure if both builds will deploy to github releases, can you take a look?

closes #65

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,9 @@ language: rust
 
 matrix:
   include:
-    - rust: nightly
+    - os: osx
+      rust: nightly-2018-05-15
+    - rust: nightly-2018-05-15
```
Comment (ah-) [inline .travis.yml]: Do we still need the version pinning?

File: .travis.yml [hunk 2]
```diff
@@ -47,6 +47,10 @@ before_deploy:
   - git config --local user.email "andreas@heider.io"
   - git tag "$(date +'%Y-%m-%d')-$TRAVIS_BUILD_NUMBER-${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH}-$(git log --format=%h -1)"
 
+branches:
+  except:
+    - /^d+-d+-d+-.*/
```
Comment (hdhoang) [inline .travis.yml]: whoops, I forgot how to regex! travis is still buildng the tagged release

File: .travis.yml [hunk 3]
```diff
@@ -42,6 +42,7 @@ before_cache:
   - chmod -R a+r $HOME/.cargo
 
 before_deploy:
+  - "if [ ${TRAVIS_OS_NAME} = 'osx' ]; then echo 'Will not tag release in macOS'; exit 1; fi"
```
Comment (hdhoang) [inline .travis.yml]: This is marking macOS as failing, thus fails the overall build. I'll find another way

Comment (hdhoang) [reply .travis.yml]: https://docs.travis-ci.com/user/deployment#Conditional-Releases-with-on%3A

one of these should help

**Pull request merged**